src/profilers/wall-profiler.ts

```diff
@@ -0,0 +1,44 @@
+import {perftools} from '../profile';
+import {serializeWallProfile} from './profile-serializer';
+const wallProfiler = require('bindings')('wall_profiler');
+
+export class WallProfiler {
+  // Average time in microseconds between samples.
+  samplingInterval: number;
+
+  // True when the profiler is actively profiling.
+  private profiling: boolean;
+
+  constructor(samplingInterval: number) {
+    this.samplingInterval = samplingInterval;  // us
+    this.profiling = false;
+    wallProfiler.setSamplingInterval(this.samplingInterval);
+  }
+
+  // Collects a profile for the duration, in milliseconds, specified by
+  // profileDuration.
+  // Returns a promise which will resolve to the collected profile.
```

Comment:
async functions always return a promise, so this description is overly wordy. `Returns the .. profile ..` would be appropriate wording here IMO.